h/models/user.py

```diff
@@ -18,27 +19,74 @@
 PASSWORD_MIN_LENGTH = 2
 
 
+def _normalise_username(username):
+    # We normalize usernames by dots and case in order to discourage attempts
+    # at impersonation.
+    return sa.func.lower(sa.func.replace(username,
+                                         sa.text("'.'"),
+                                         sa.text("''")))
+
+
+class UsernameComparator(Comparator):
```

Comment:
Am I right in surmising that the reason you need to implement this is a full-on `Comparator` rather than using the more concise `@username.expression` is so that it’ll also convert the other side of the lookup? So that something like `query(User).filter(User.username == 'Enterprise.Synergy')` will still work?

I think that’s what you mean in the comment, but I just wanted to check my understanding that `@username.expression` wouldn’t do that for you.